src/tscpp/util/unit_tests/test_TextView.cc

```diff
@@ -252,6 +252,59 @@ TEST_CASE("TextView Affixes", "[libts][TextView]")
   s = "file.cc.org";
   s.remove_prefix('!');
   REQUIRE(s.empty());
+
+  // From MIMEHdr::get_host_port_values
```

Comment:
Shouldn't we have a test for `MIMEHdr::get_host_port_values`? How do we guarantee the logic here and there are the same?

Reply:
Textual comparison. I literally cut and pasted the code. That's sufficient for this transition. Future changes should have their own tests independent of `TextView`. It's not possible for this unit test to reach back to `MIMEHdr`.

Reply:
Do you mean one time comparison on this PR is enough and those logic can be different in the future? If so, this comment line may be misleading. And I'm not sure what the point of this test is. Maybe current tests for find, size, take_prefix_at and split_prefix_at are not enough?

Reply:
The logic might be different in the future, and that can be handled then. These tests are being extra cautious because of the switch in the underlying classes. This isn't just testing `TextView` but the new logic in `MIMEHdr` that uses `TextView`. I can take the tests out if you want.

Reply:
> The logic might be different in the future, and that can be handled then.

How? I'm not going to remember there is a dup code here.

> These tests are being extra cautious because of the switch in the underlying classes. This isn't just testing TextView but the new logic in MIMEHdr that uses TextView.

That's the reason I suggested to test `MIMEHdr::get_host_port_values`. Why can't we create a MIMEHdr object that has Host header and call `get_host_port_values`?

>I can take the tests out if you want.

Having more tests is good in general, but I don't think I can maintain this test because I don't understand the point.

Reply:
Handled by writing tests for the change when the change is made. I am unable to anticipate what that might be before it happens.

We can't create a `MIMEHdr` object in this test because it doesn't link with those libraries and doing so would create circular dependencies.

I don't see why this test would have to be maintained. If the `MIMEHdr` logic doesn't change, no problem. If it does, who ever changes it can write tests to verify it at that point, even if they have no idea these tests exist. Because this change, in theory, does not change how the `MIMEHdr` logic works, the existing `MIMEHdr` tests should suffice. These tests are simply extra caution.

Reply:
> We can't create a MIMEHdr object in this test because it doesn't link with those libraries and doing so would create circular dependencies.

This is because you added the test in inappropriate place.

> If the MIMEHdr logic doesn't change, no problem. If it does, who ever changes it can write tests to verify it at that point, even if they have no idea these tests exist. 

The logic is not everything. You should make sure that the behavior is not changed.

> the existing MIMEHdr tests should suffice. 

Where is that?

Reply:
Existing tests are in "proxy/hdrs/unit_tests".

Reply:
None of the tests in the directory tests get_host_port_values. That's why I had to wrote 6063.